main.nf

```diff
@@ -8,269 +8,4129 @@
+               ipo_min_peakwidth_h=!{params.ipo_min_peakwidth_l_pos} ipo_max_peakwidth_l=!{params.ipo_max_peakwidth_l_pos} ipo_max_peakwidth_h=!{params.ipo_max_peakwidth_h_pos} ipo_ppm_l=!{params.ipo_ppm_l_pos} \
+             ipo_ppm_h=!{params.ipo_ppm_h_pos} ipo_mzdiff_l=!{params.ipo_mzdiff_l_pos} ipo_mzdiff_h=!{params.ipo_mzdiff_h_pos} ipo_charge_camera=!{params.ipo_charge_camera_pos} ipo_max_ppm_camera=!{params.ipo_max_ppm_camera_pos} \
+             response_l=!{params.ipo_response_l_pos} response_h=!{params.ipo_response_h_pos} distFunc=!{params.ipo_distFunc_pos} factorDiag_l=!{params.ipo_factorDiag_l_pos} factorDiag_h=!{params.ipo_factorDiag_h_pos} factorGap_l=!{params.ipo_factorGap_l_pos} \
+             factorGap_h=!{params.ipo_factorGap_h_pos} localAlignment=!{params.ipo_localAlignment_pos} ipo_gapInit_l=!{params.ipo_gapInit_l_pos} ipo_gapInit_h=!{params.ipo_gapInit_h_pos} ipo_gapExtend_l=!{params.ipo_gapExtend_l_pos} \
+             ipo_gapExtend_h=!{params.ipo_gapExtend_h_pos} ipo_profStep_l=!{params.ipo_profStep_l_pos} ipo_profStep_h=!{params.ipo_profStep_h_pos} bw_l=!{params.ipo_bw_l_pos} bw_h=!{params.ipo_bw_h_pos} minfrac_l=!{params.ipo_minfrac_l_pos} \
+              minfrac_h=!{params.ipo_minfrac_h_pos} mzwid_l=!{params.ipo_mzwid_l_pos} mzwid_h=!{params.ipo_mzwid_h_pos} minsamp_l=!{params.ipo_minsamp_l_pos} \
+               minsamp_h=!{params.ipo_minsamp_h_pos} max_l=!{params.ipo_max_l_pos} max_h=!{params.ipo_max_h_pos} ncores=!{params.ipo_ncores_pos} outputxset=quant_params_pos.json outputrt=rt_params_pos.json
+          """
+      }
+      }
+
+        param_target_to_detection_process_pos= ( ipo_pos_globalQ
+                     ? param_to_detection_process_pos
+                     :  file("NO_QFILE"))
+                     param_target_to_rt_process_pos= ( ipo_pos_globalAvoidRT==true
+                                  ? file("NO_RTFILE")
+                                  :  param_to_rt_process_pos)
+
+     process process_masstrace_detection_pos_xcms_centroided{
+       label 'xcms'
+       tag "$name"
+       publishDir "${params.outdir}/process_masstrace_detection_pos_xcms", mode: params.publish_dir_mode, enabled: params.publishDir_intermediate
+
+
+
+       input:
+       file mzMLFile from masstrace_detection_process_pos
+       each file(phenotype_file) from phenotype_design_pos
+       each file(paramsQ) from param_target_to_detection_process_pos
+
+       output:
+       file "${mzMLFile.baseName}.rdata" into collect_rdata_pos_xcms
+       file "${mzMLFile.baseName}.mzML" into rt_rdata_pos_xcms
+
+       shell:
+        def filter_argument = paramsQ.name != 'NO_QFILE' ? "ipo_in=${paramsQ}" : ''
+       """
+    /usr/local/bin/findPeaks.r input=\$PWD/!{mzMLFile} output=\$PWD/!{mzMLFile.baseName}.rdata ppm=!{params.masstrace_ppm_pos_xcms} peakwidthLow=!{params.peakwidthlow_quant_pos_xcms} \
```

Comment:
```suggestion
    /usr/local/bin/findPeaks.r input=\$PWD/!{mzMLFile} \
                                              output=\$PWD/!{mzMLFile.baseName}.rdata ppm=!{params.masstrace_ppm_pos_xcms} \
                                              peakwidthLow=!{params.peakwidthlow_quant_pos_xcms} \
```

Reply:
I think you might need double-slashes for it to get through Nextflow parsing nicely (otherwise the final command in the output just has huge chunks of whitespace, that makes it difficult to read in the pipeline report etc).

Reply:
Fixed